src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ActionPlanService.kt

```diff
@@ -47,10 +47,14 @@ class ActionPlanService(
       ?: throw EntityNotFoundException("draft action plan not found [id=$id]")
   }
 
-  fun updateActionPlan(update: ActionPlan): ActionPlan {
-    val draftActionPlan = getDraftActionPlan(update.id)
-    actionPlanValidator.validateDraftActionPlanUpdate(update)
-    updateDraftActivityPlan(draftActionPlan, update)
+  fun updateActionPlan(
+    actionPlanId: UUID,
+    numberOfSessions: Int?,
+    activityUpdate: ActionPlanActivity?,
+  ): ActionPlan {
+    val draftActionPlan = getDraftActionPlan(actionPlanId)
+    updateDraftActivityPlan(draftActionPlan, numberOfSessions, activityUpdate)
+    actionPlanValidator.validateDraftActionPlanUpdate(draftActionPlan)
```

Comment:
shouldn't we validate before updating? then if it's invalid we save a few cycles.

Reply:
The validator is an ActionPlan validator as apposed a request object validator which means it can only be done once the update has been applied to the entity object. I do take your point but the validator has greater reuse as an entity validator.